### DISCUSSION TITLE
I want to recompile the 0.12.2 version of vevtor, how can I see which version of rust it is using?

### DISCUSSION THREAD

**Discussion by FengZh61** [vectordotdev/vector#16509] [Q&A] [answered]
#16361

Vector-0.12.2-x86_64-unknown-linux-musl can run on Linux 5.7 with kernel version 2.6.18-274.el5.

But it runs with a little error report in use host_metrics.

Feb 20 11:27:16.730 ERROR source{component_kind="source" component_name=host component_type=host_metrics}: vector::sources::host_metrics: Internal log [Failed to load memory info.] is being rate limited.
Feb 20 11:27:16.731 ERROR source{component_kind="source" component_name=host component_type=host_metrics}: vector::sources::host_metrics: Internal log [Failed to load/parse network data.] is being rate limited.

I want to recompile it.

So which version of rust should I use, or in which file is the information about the rust version recorded?

**Comment by jszwedko** [comment, marked as answer]
Hi @FengZh61 ! You can see which version of Rust was used by looking at the `rust-toolchain` file. In this case, it looks like 1.50.0: https://github.com/vectordotdev/vector/blob/v0.12.2/rust-toolchain

**Reply by FengZh61** [reply]
> Hi @FengZh61 ! You can see which version of Rust was used by looking at the `rust-toolchain` file. In this case, it looks like 1.50.0: https://github.com/vectordotdev/vector/blob/v0.12.2/rust-toolchain

Thanks~
